src/planner/binder/query_node/bind_select_node.cpp

```diff
@@ -1,38 +1,45 @@
+#include "duckdb/common/limits.hpp"
+#include "duckdb/common/string_util.hpp"
+#include "duckdb/execution/expression_executor.hpp"
+#include "duckdb/main/config.hpp"
 #include "duckdb/parser/expression/columnref_expression.hpp"
 #include "duckdb/parser/expression/comparison_expression.hpp"
 #include "duckdb/parser/expression/constant_expression.hpp"
+#include "duckdb/parser/expression/subquery_expression.hpp"
+#include "duckdb/parser/expression/table_star_expression.hpp"
 #include "duckdb/parser/query_node/select_node.hpp"
 #include "duckdb/parser/tableref/joinref.hpp"
 #include "duckdb/planner/binder.hpp"
-#include "duckdb/main/config.hpp"
-#include "duckdb/execution/expression_executor.hpp"
 #include "duckdb/planner/expression_binder/constant_binder.hpp"
 #include "duckdb/planner/expression_binder/group_binder.hpp"
 #include "duckdb/planner/expression_binder/having_binder.hpp"
 #include "duckdb/planner/expression_binder/order_binder.hpp"
 #include "duckdb/planner/expression_binder/select_binder.hpp"
 #include "duckdb/planner/expression_binder/where_binder.hpp"
 #include "duckdb/planner/query_node/bound_select_node.hpp"
-#include "duckdb/parser/expression/table_star_expression.hpp"
-#include "duckdb/common/limits.hpp"
-#include "duckdb/common/string_util.hpp"
+
+#include <duckdb/planner/expression_binder/aggregate_binder.hpp>
 
 namespace duckdb {
 
-static int64_t BindConstant(Binder &binder, ClientContext &context, string clause, unique_ptr<ParsedExpression> &expr) {
+static bool BindConstant(Binder &binder, ClientContext &context, const string& clause, unique_ptr<ParsedExpression> &expr, int64_t& delimiter) {
 	ConstantBinder constant_binder(binder, context, clause);
```

Comment:
Instead of doing try/catch using the constant binder, just bind using the `ExpressionBinder` and afterwards check if the result is constant (by checking `bound_expr->IsFoldable()`). We want to avoid using try/catch for regular control flow (i.e. code inside the engine should preferably never use try/catch, throwing an exception should always abort the query).